module/documents/actor/bastion.mjs

```diff
@@ -284,11 +284,16 @@ export default class Bastion {
       if ( (trade.pending.value === null) && trade.pending.stocked ) updates["system.trade.stock.stocked"] = true;
 
       // Bought goods
-      else if ( trade.pending.value !== null && !trade.pending.creatures.length ) {
-        updates["system.trade.stock.value"] = Math.min(trade.stock.value + trade.pending.value, trade.stock.max);
+      else if ( trade.pending.value !== null ) {
+        if ( trade.pending.creatures.length ) {
+          const creatures = trade.creatures.value.concat(trade.pending.creatures);
+          const diff = trade.creatures.max - creatures.length;
+          if ( diff < 0 ) creatures.splice(diff);
```

Comment:
I think this logic is a bit off, because if you are buying 2 creatures the max is `1` (`3` normal - `2` pending) and so this would be negative, resulting in only one creature being purchased.

Reply:
Oops, thank you. Forgot to edit this after making the data prep changes.

Reply:
Just removed this check. Since the `max` is a function of `pending`, and `value` can't be greater than `max`, any arithmetic we do here is just redundant.